Readme (#10)
webex/components: README.md

## Table of Contents
- [Install](#install)
- [Usage](#usage)
- [Contributing](#contributing)
- [License](#license)
- [Support](#support)
- [Team](#team)

## Install

```bash
npm i -S @webex/components @momentum-ui/react
```

## Usage
**Webex Components** are built with [Momentum UI](https://github.com/momentum-design/momentum-ui) React components. We need to externally import all the fonts, icons, images and core *CSS* manually. There are two ways to do this:

### 1. CSS
1. Copy the `@momentum-ui/core/fonts` directory to the 'fonts' directory for your site.
2. Copy the `@momentum-ui/core/images` directory to the 'images' directory for your site.
3. Copy the `@momentum-ui/core/css/momentum-ui.min.css` to your styles/css directory.
4. In the `<head>`, of your HTML reference the location of your momentum-ui.min.css:

    `<link rel='stylesheet' href='path/to/momentum-ui/css/momentum-ui.min.css'>`

***
### 2. SCSS
1. Copy the `@momentum-ui/icons/fonts` directory to the `fonts` directory for your site.

> 2. Add an `$brand-font-folder`, `$icon-font-path` variables to your app's variables.scss file.

Reviewer: ```suggestion
2. Add `$brand-font-folder` and `$icon-font-path` variables to the top of your app's sass file.
```

Since we don't really refer to `variables.scss` anywhere.

<h1 align='center' style='border-bottom: none;'>Webex Components</h1>
<h3 align='center'>Library of React components to easily embed into your web applications!</h3>
<p align='center'>
<a href='https://circleci.com/gh/webex/components'>
    <img alt='CircleCI' src='https://circleci.com/gh/webex/components.svg?style=shield'>
  </a>
  <a href='#badge'>

>     <img alt='semantic-release' src='https://img.shields.io/badge/%20%20%F0%9F%93%A6%F0%9F%9A%80-semantic--release-e10079.svg'>

Reviewer 2: What is this badge showing?

  </a>
  <a href='#badge'>
    <img alt='semantic-release' src='https://img.shields.io/badge/%20%20%F0%9F%93%A6%F0%9F%9A%80-semantic--release-e10079.svg'>
  </a>
  <a href='https://github.com/webex/components/blob/master/package.json#L28'>
    <img src='https://img.shields.io/npm/l/webex.svg' alt='license'>
  </a>
</p>

**Webex Components** is a set of [React](https://reactjs.org) components following Webex standard styling, aimed at react developers that want to embed the components into their applications.

## Table of Contents
- [Install](#install)
- [Usage](#usage)
- [Contributing](#contributing)
- [License](#license)
- [Support](#support)
- [Team](#team)

## Install

```bash
npm i -S @webex/components @momentum-ui/react
```

## Usage
**Webex Components** are built with [Momentum UI](https://github.com/momentum-design/momentum-ui) React components. We need to externally import all the fonts, icons, images and core *CSS* manually. There are two ways to do this:

### 1. CSS

> 1. Copy the '@momentum-ui/core/fonts' directory to the 'fonts' directory for your site.

Reviewer 2: Change all the `'` to ```

Author: oh okay

2. Copy the '@momentum-ui/core/images' directory to the 'images' directory for your site.
3. Copy the '@momentum-ui/core/css/momentum-ui.min.css' to your styles/css directory.
4. In the `<head>`, of your HTML reference the location of your momentum-ui.min.css:

    `<link rel='stylesheet' href='path/to/momentum-ui/css/momentum-ui.min.css'>`

***
### 2. SCSS
1. Copy the '@momentum-ui/icons/fonts' directory to the 'fonts' directory for your site.
2. Add an '$brand-font-folder, $icon-font-path variables to your app's variables.scss file.

    ```scss
    $brand-font-folder: "~@momentum-ui/core/fonts";
    $icon-font-path: "~@momentum-ui/icons/fonts";
    ```

3. Import `@momentum-ui/core/scss/momentum-ui-components` into your main entry Scss file AFTER the variables.

    ```scss

>     @import '~@momentum-ui/core/scss/momentum-ui-components';

Reviewer 2: Keep the quotations consistent with the previous text, use `"` or `'`

Author: I stick with double quotes on README files.

    ```

4. Compile Sass files using your static compiler or bundler (Webpack, Grunt, Gulp, etc.)
***

Once Momentum styles have been imported into your project, you can start using the components. For example:

  ```js
  import {WebexAvatar} from '@webex/components';

  function App() {
    return (
      <div className='App'>
        <WebexAvatar title='Webex Avatar' />
      </div>
    );
  }

  export default App;
  ```

*Happy Coding!*

## Contributing
We'd love for you to contribute to our source code and to make Webex Components even better than it is today! Here are some [guidelines](https://github.com/webex/components/blob/master/CONTRIBUTING.md) that we'd like you to follow.

### Issues

> Please open an [issue](https://github.com/webex/components/issues) and we will get to it in an orderly manner. Please leave as much as information as possible for a better underestanding.

Reviewer: `understanding` is misspelled as `underestanding`.

### Contributing Code
We are using [Webex Style Guide eslint rule](https://github.com/webex/web-styleguide/tree/master/packages/node_modules/%40webex/eslint-config-react) and [prettier](https://github.com/prettier/prettier) to lint the code style.
You can "prettify" your code before committing via:

  ```bash
  npm run prettier:write
  ```

### Release Process

> There is a list of commit types provided [here](https://github.com/webex/components/blob/master/CONTRIBUTING.md#type). However, not all commits triggers our release process. We are using [semantic-release](https://github.com/semantic-release/semantic-release) to fully automate the version management and package publishing. By default `semantic-release` uses the [Angular Commit Message Conventions](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines) and triggers release and publishing based on the following rules:

Reviewer: `not all commits trigger our release process`

> <center>

Reviewer 2: What do this `<center>` tags do? The table it's not really being centered, and I think it's ok that way

Author: okay

  </a>
</p>

**Webex Components** is a set of [React](https://reactjs.org) components following Webex standard styling, aimed at react developers that want to embed the components into their applications.

## Table of Contents
- [Install](#install)
- [Usage](#usage)
- [Contributing](#contributing)
- [License](#license)
- [Support](#support)
- [Team](#team)

## Install

```bash
npm install --save @webex/components @momentum-ui/react
```

## Usage
**Webex Components** are built with [Momentum UI](https://github.com/momentum-design/momentum-ui) React components. We need to externally import all the fonts, icons, images and core *CSS* manually. There are two ways to do this:

### 1. CSS
1. Copy the `@momentum-ui/core/fonts` directory to the 'fonts' directory for your site.
2. Copy the `@momentum-ui/core/images` directory to the 'images' directory for your site.
3. Copy the `@momentum-ui/core/css/momentum-ui.min.css` to your styles/css directory.
4. In the `<head>`, of your HTML reference the location of your momentum-ui.min.css:

    `<link rel='stylesheet' href='path/to/momentum-ui/css/momentum-ui.min.css'>`

***
### 2. SCSS

> 1. Copy the `@momentum-ui/icons/fonts` directory to the `fonts` directory for your site.

Reviewer: Since we are importing from node_modules, this step isn't needed any more.

    ```scss
    $brand-font-folder: "~@momentum-ui/core/fonts";
    $icon-font-path: "~@momentum-ui/icons/fonts";
    ```

> 2. Import `@momentum-ui/core/scss/momentum-ui-components` into your main entry Scss file AFTER the variables.

Reviewer 2: Sass

<h1 align='center' style='border-bottom: none;'>Webex Components</h1>
<h3 align='center'>Library of React components to easily embed into your web applications!</h3>
<p align='center'>
<a href='https://circleci.com/gh/webex/components'>
    <img alt='CircleCI' src='https://circleci.com/gh/webex/components.svg?style=shield'>
  </a>
  <a href='#badge'>
    <img alt='semantic-release' src='https://img.shields.io/badge/%20%20%F0%9F%93%A6%F0%9F%9A%80-semantic--release-e10079.svg'>
  </a>
  <a href='https://www.npmjs.com/package/@webex/components'>
    <img alt='npm latest version' src='https://img.shields.io/npm/v/@webex/components?label=npm%40latest'>
  </a>
  <a href='https://github.com/webex/components/blob/master/package.json#L28'>
    <img src='https://img.shields.io/npm/l/webex.svg' alt='license'>
  </a>
</p>

> **Webex Components** is a set of *React* components following Webex Teams standard styling, aimed at react developers that want to embed the components into their applications.

Author: that will be added once we have the work implemented.

> **Webex Components** is a set of *React* components following Webex Teams standard styling, aimed at react developers that want to embed the components into their applications.

Reviewer 2: Sure

## Table of Contents
- [Install](#install)
- [Usage](#usage)
- [Contributions](#contributions)
- [License and Support](#license)

<a name='install'></a>
## Install

```bash

> npm i -S @webex/components @momentum-ui/react

Author: it is not needed.

> npm i -S @webex/components @momentum-ui/react

Reviewer 2: Can you just spell it out then? `install --save`?

> npm i -S @webex/components @momentum-ui/react

Author: why?

> npm i -S @webex/components @momentum-ui/react

Reviewer: I think `install --save` is more readable for users who might not be familiar with npm commands.

> npm i -S @webex/components @momentum-ui/react

Author: :|

> npm i -S @webex/components @momentum-ui/react

Author: Okay

```

<a name='usage'></a>
## Usage

> **Webex Components** are using *Cisco style standard* presentational  [momentum ui](https://github.com/momentum-design/momentum-ui) react components. And because of that we need to import all the fonts, icons, images and core *CSS* manually. There are two ways to do this:

Reviewer 2: `...react components. And because of...` is grammatically improper. You can rephrase the sentence as:

`
Webex components use Webex presentational style, from Momentum UI's design system. Momentum UI's fonts, icons, images and styling must be imported separately from the components. There are two ways to do this:
`

Author: I addressed this already

### 1. CSS
1. Copy the '@momentum-ui/core/fonts' directory to the 'fonts' directory for your site.
2. Copy the '@momentum-ui/core/images' directory to the 'images' directory for your site.
3. Copy the '@momentum-ui/core/css/momentum-ui.min.css' to your styles/css directory.
4. In the `<head>` of your HTML reference the location of your momentum-ui.min.css.

    `<link rel='stylesheet' href='path/to/momentum-ui/css/momentum-ui.min.css'>`

***
### 2. SCSS
1. Copy the '@momentum-ui/icons/fonts' directory to the 'fonts' directory for your site.
2. Add an '$brand-font-folder, $icon-font-path, and $images-path' variables to your app's variables.scss file.

    ```scss
    $brand-font-folder: 'path/to/fonts/directory';
    $icon-font-path: 'path/to/fonts/directory';
    $images-path: 'path/to/images/directory';
    ```

3. Import '@momentum-ui/core/scss/momentum-ui.scss' into your main entry Scss file AFTER the variables.

    `@import '~@momentum-ui/core/scss/momentum-ui';`

> 4. Compile your Scss using your static compiler or bundler (Webpack, Grunt, Gulp, etc.)

Reviewer 2: `Compile Sass files using ...`

***

> Now that we have the targeted style library imported into our application, we can go ahead and import the components as it shows in the following code:

Reviewer 2: The first part of the sentence is a bit confusing, you can rephrase as:

`
Once Momentum styles have been imported into your project, you can start using the components. For example:
`

  ```js
  import {WebexAvatar} from '@webex/components';

>   import './App.css';

Reviewer 2: Where is this coming from?

  function App() {
    return (
      <div className='App'>
        <WebexAvatar title='Webex Avatar' />
      </div>
    );
  }

  export default App;
  ```

*Happy Coding!*

<a name='contributions'></a>

> ## Contributions

Reviewer: Should we just link to the CONTRIBUTING document?

> ## Contributions

Reviewer 2: There's a missing link to CONTRIBUTING.md on this section

Author: addressed

> ## Contributions

Reviewer 2: There's a section missing on how to open PRs/fork, etc. We can merge this as-is and open another PR to add the section, though.

Author: its pretty obvious how to open a PR

Author: you are welcome to open a PR for it lalli :)

Reviewer 2: Is there a reason to not include in in this PR now?

Author: yes, I don't think its necessary to do that right now. I have other priorities. And we addressed this in the space. You can go ahead and create a JIRA for it and we can prioritize it later.

> As much as we truly appreciate your contributions to this projects, there are some rules that we expect all of our contributors to follow.

Reviewer 2: I like how [angular.js](https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#contributing-to-angularjs) has phrased their contribution intro:
`
We'd love for you to contribute to our source code and to make AngularJS even better than it is today! Here are the guidelines we'd like you to follow:
`

Author: sure lalli

### Issues

> Please open an [issue](https://github.com/webex/components/issues) and we will get to it in an orderly manner. Please leave as much as information as possible for a better underestanding.

Reviewer 2: This is a good start, we should probably create an issue to setup a more explicit issue-handling process.

### Coding

> We are using [webex standard eslint](https://github.com/webex/web-styleguide) and [husky](https://github.com/typicode/husky) to lintify the code style. Please make sure to prettify your code before committing.

Reviewer: ```suggestion
We are using [Webex Style Guide eslint rule](https://github.com/webex/web-styleguide/tree/master/packages/node_modules/%40webex/eslint-config-react) and [prettier](https://github.com/prettier/prettier) to lint the code style. 

You can "prettify" your code before committing via:
```

Husky doesn't lint the code, just runs the linters.

  ```bash
  npm run prettier:write
  ```

### Release Process
We are using [semantic-release](https://github.com/semantic-release/semantic-release) to fully automate the version management and package publishing. By default `semantic-release` uses the [Angular Commit Message Conventions](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines) and triggers releases based on the following rules:

<center>

|Commit|Release type|
|------|------------|

> |Commit with breaking change| Major release|

Reviewer: Could you add a bit more about how to trigger a breaking change?

|Commit with type `feat`| Minor release|
|Commit with type `fix` | Patch release|
|Commit with type `perf`| Patch release|
</center>

#### Commit linter
We are using [Commitlint](https://github.com/conventional-changelog/commitlint) and [husky](https://github.com/typicode/husky) to lintify the commit messages. Please make sure to choose the appropriate commit `type`, `scope` and `description`. More information [here](https://github.com/webex/components/blob/master/CONTRIBUTING.md).

<a name='license'></a>
## License and Support

> *MIT* License.

Reviewer 2: Link to the license file

### Support

> Please click [here](https://developer.webex.com/support) to choose the appropriate method for more help.

Reviewer: ```suggestion
Developer support is available via the Webex for Developers portal: https://developer.webex.com/support
```

> Please click [here](https://developer.webex.com/support) to choose the appropriate method for more help.

Reviewer 2: Explicitly state "here" as the developer portal.
`
For more developer resources, tutorials and support, visit the Webex developer portal, https://developer.webex.com
`

> ### Owner

Reviewer 2: I don't think is necessary to have an "owner" section externally. From a contribution/developer point of view is a bit irrelevant.
